### PR TITLE
fix(language-service): getSourceFile() should only be called on TS files

### PR DESCRIPTION
**Pull request by kyliau**
This is in preparation for Angular LS plugin handling external templates (HTML) files.
`getSourceFile()` parses the file and returns `SourceFileObject`, so it must never be called on HTML.
Otherwise, TS errors would show up in diagnostics.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/language-service/src/typescript_host.ts [hunk 1]
```diff
@@ -117,18 +117,25 @@ export class TypeScriptServiceHost implements LanguageServiceHost {
...
+        this.context = sourceFile.fileName;
+        const node = this.findNode(sourceFile, position);
+        if (node) {
+          return this.getSourceFromNode(
+              fileName, this.host.getScriptVersion(sourceFile.fileName), node);
+        }
       }
     } else {
       this.ensureTemplateMap();
-      // TODO: Cannocalize the file?
```
Comment (ayazhafiz) [inline packages/language-service/src/typescript_host.ts]: Is this TODO now completed?

File: packages/language-service/src/typescript_host.ts [hunk 2]
```diff
@@ -191,10 +191,22 @@ export class TypeScriptServiceHost implements LanguageServiceHost {
         ts.forEachChild(sourceFile, visit);
       }
       return result.length ? result : undefined;
+    } else {
+      this.ensureTemplateMap();
+      const componentType = this.fileToComponent.get(fileName);
```
Comment (ayazhafiz) [inline packages/language-service/src/typescript_host.ts]: nit: prefer `componentSymbol` or similar, `componentType` is ambiguous - are there different kinds of components? Is this the component class type?

File: packages/language-service/src/typescript_host.ts [hunk 3]
```diff
@@ -212,6 +224,9 @@ export class TypeScriptServiceHost implements LanguageServiceHost {
   }
 
   getSourceFile(fileName: string): ts.SourceFile|undefined {
+    if (!fileName.endsWith('.ts')) {
+      throw new Error(`Non-TS source file requested: ${fileName}`);
+    }
     return this.tsService.getProgram() !.getSourceFile(fileName);
```
Comment (ayazhafiz) [inline packages/language-service/src/typescript_host.ts]: Can you do `this.getSourceFile(fileName)` here?

File: packages/language-service/src/typescript_host.ts [hunk 4]
```diff
@@ -117,18 +117,25 @@ export class TypeScriptServiceHost implements LanguageServiceHost {
     return this.templateReferences || [];
   }
 
+  /**
```
Comment (ayazhafiz) [inline packages/language-service/src/typescript_host.ts]: Are these changes testable?

**Review by ayazhafiz — CHANGES_REQUESTED** [review]

File: packages/language-service/src/typescript_host.ts (hunk 3 above)
Comment (kyliau) [reply packages/language-service/src/typescript_host.ts]: ? That'll be infinite recursion...

File: packages/language-service/src/typescript_host.ts (hunk 2 above)
Comment (kyliau) [reply packages/language-service/src/typescript_host.ts]: done.

Comment (kyliau) [reply packages/language-service/src/typescript_host.ts]: It's really an identifier ([StaticSymbol](https://github.com/angular/angular/blob/master/packages/compiler/src/aot/static_symbol.ts)) for the component that owns the external template.

File: packages/language-service/src/typescript_host.ts (hunk 4 above)
Comment (kyliau) [reply packages/language-service/src/typescript_host.ts]: Added a simple test.

File: packages/language-service/src/typescript_host.ts (hunk 1 above)
Comment (kyliau) [reply packages/language-service/src/typescript_host.ts]: The TODO is out of date and is no longer needed.

**Review by ayazhafiz — APPROVED** [review]
lgtm

**Comment by kyliau** [comment]
caretaker: test_docs_examples_ivy failure is unrelated to this change

**Pull request closed without merging**